Onboarding/Connector/Scripts/onboard-subscription-for-impact-reporting.sh

```diff
@@ -0,0 +1,256 @@
+
+  role_id=$(az role definition list -n "$role_name"  --query [].id -o tsv)
+  if [ ! "$role_id" ]; then
+    log "Creating custom role for alerts reading."
+    
+    custom_role_json_file="/tmp/custom-role.json"
+    create_custom_role_json_file "$subscription_id" "$role_name" "$custom_role_json_file"
+    role_id=$(az role definition create --role-definition "$custom_role_json_file" | jq -r ".id")
+    rm -rf "$custom_role_json_file"
+    
+    log "Custom role: $role_name for alerts reading created successfully with role id: $role_id."
+  else
+    log "Custom role: $role_name for alerts reading already exists with role id: $role_id."
+  fi
+  
+  log "Assigning the custom role: $role_name to the connector app: $connector_app_name"
+  connector_principal_id=$(az ad sp list --display-name "$connector_app_name" --query "[].id" -o tsv)
+
+  assigned_custom_role=$(az role assignment list --assignee "$connector_principal_id" --role "$role_id" --query [].id -o tsv)
+
+  if [ ! "$assigned_custom_role" ]; then
+    az role assignment create --role "$role_id" --assignee-object-id "$connector_principal_id" --scope "/subscriptions/$subscription_id"
+    log "Custom role: $role_name assigned to the connector app: $connector_app_name"
+  else
+    log "Custom role: $role_name already assigned to the connector app: $connector_app_name"
+  fi
+
+  log "Successfully setup permissions for alerts reading" 
+}
+
+deploy_connector()
+{
+  subscription_id=$1
+  connector_name="AzureMonitorConnector"
+  log "Deploying connector: $connector_name for impact reporting" 
+
+  request_body="{
+      \"properties\": \
+          {\"connectorType\": \"AzureMonitor\"\
+      }\
+    }"
+  url="https://management.azure.com/subscriptions/$subscription_id/providers/Microsoft.Impact/connectors/$connector_name?api-version=2024-05-01-preview"
+
+  attempt_count=1
+  sleep_duration=10
+
+  while ! az rest --method put --url "$url" --body "$request_body" --query "properties.provisioningState" | grep -q "Succeeded"; do
+    sleep_duration=$((sleep_duration*2))
+    log "Attempt #$attempt_count: Connector deployment is in progress. Waiting for $sleep_duration seconds before retrying."
+    sleep "$sleep_duration"
+    attempt_count=$((attempt_count+1))
+  done 
+
+  log "Deployment of connector: $connector_name is complete" 
+}
+
+validate_input()
+{
+  [[ -z "$subscription_id" && -z "$file_path" ]] && (log "Please provided required input: subscription_id or file_path with list of subscription IDs while invoking the script" && exit 1)
+  [[ -n "$subscription_id" && -n "$file_path" ]] && (log "Please provided either subscription_id or file_path with list of subscription IDs while invoking the script. Do not provide both." && exit 1)
+
+  [[ -n "$subscription_id" ]] && (log "Using Subscription ID = $subscription_id")
+  
+  if [[ -n "$file_path" ]]; then
+    log "Subscription IDs will be read from the file: $file_path"
+    if [[ ! -f "$file_path" ]]; then
```

Comment:
this log line: log "Subscription IDs will be read from the file: $file_path" is to indicate it will be used, the other check is for ensuring the path exists from where we are reading.
I can flip it if it makes more sense that way.

Reply:
Yeah, the flipping is what I meant. Functionally it's the same, but if the file doesn't exist, then we should just say that and not print any log to give an indication of "okay, we'll be reading from this file".